google/auth/_exponential_backoff.py

```diff
@@ -86,7 +88,7 @@ def __iter__(self):
 
     def __next__(self):
         if self._backoff_count >= self._total_attempts:
-            raise StopIteration
+            raise exceptions.StopIteration
```

Comment:
This doesn't seem necessary. 

I think the for loop catches the `StopIteration`, so it should never leave the loop block

Reply:
I didn't quite understand this part.
Is it in the
```
for _ in retries:
   ...
```